### PR TITLE
add 19-candy-store-tests example

### PR DESCRIPTION
**Pull request by azizi-a**

### CODE REVIEW

File: step-by-step/18-hello-test/README.md [hunk 1]
```diff
@@ -222,6 +229,13 @@ assertion is correct we can return an `Ok` (containing an empty value):
     Ok(())
 ```
 
+If you're wondering about the new syntax, Scrypto-Test uses
+[`Result`](https://doc.rust-lang.org/std/result/) return types for error
+handling, so we can use the `?` operator to propagate errors up the call stack,
+and OK to return function the values. In our case just we're returning `Ok(())`
```
Comment (motosharpley) [inline step-by-step/18-hello-test/README.md:235]: and OK to return the function values.

File: step-by-step/19-candy-store-tests/README.md [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+  - [Unit Testing the Gumball Machine Module](#unit-testing-the-gumball-machine-module)
+  - [Integration Testing the Candy Store Module](#integration-testing-the-candy-store-module)
+- [Using Candy Store Tests](#using-candy-store-tests)
+- [Closing thoughts](#closing-thoughts)
+
+## Testing the Candy Store Package
+
+We can have multiple test files for a Scrypto package by placing them in the
+`tests/` directory. We've done this with the separate modules of our package
+which are each have their own test files, `tests/candy_store.rs` and
```
Comment (motosharpley) [inline step-by-step/19-candy-store-tests/README.md:24]: which each have their own test files,

File: step-by-step/19-candy-store-tests/README.md [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+
+```rs
+#[test]
+fn can_instantiate_gumball_machine() -> Result<(), RuntimeError> {
+    let (_env, _gumball_machine) = arrange_test_environment(dec!(1))?;
+    Ok(())
+}
+```
+
+After this, the rest of our unit test follow a similar pattern. With clear
```
Comment (motosharpley) [inline step-by-step/19-candy-store-tests/README.md:100]: the rest of our unit tests

File: step-by-step/19-candy-store-tests/README.md [hunk 4]
```diff
@@ -0,0 +1,255 @@
...
+
+```toml
+[dev-dependencies]
+# --snip---
+scrypto-unit = { git = "https://github.com/radixdlt/radixdlt-scrypto", tag = "v1.1.1" }
+# --snip---
+radix-engine-interface = { git = "https://github.com/radixdlt/radixdlt-scrypto", tag = "v1.1.1" }
+```
+
+Then we add the required imports to `tests/gumball_machine.rs`:
```
Comment (motosharpley) [inline step-by-step/19-candy-store-tests/README.md:156]: is this correct should this be candys_store.rs?

File: step-by-step/19-candy-store-tests/README.md [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+    // If the transaction is unsuccessful, the test will fail here
+    receipt.expect_commit_success();
+```
+
+The
+[TestRunner](https://radixdlt.github.io/radixdlt-scrypto/scrypto_unit/struct.TestRunner.html)
+has various methods to arrange and execute in the simulated ledger, like
+`new_allocated_account()` and `execute_manifest_ignoring_fee()` while the
+[ManifestBuilder](https://docs.radixdlt.com/docs/rust-manifest-builder) is used
+to create a transaction manifests inside a rust file. These allow us to test a
```
Comment (motosharpley) [inline step-by-step/19-candy-store-tests/README.md:218]: a transaction manifest or  transaction manifests

File: step-by-step/19-candy-store-tests/README.md [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+variety of interaction with the Candy Store and be sure that none will fail.
```
Comment (motosharpley) [inline step-by-step/19-candy-store-tests/README.md:219]: interactions

File: step-by-step/19-candy-store-tests/README.md [hunk 7]
```diff
@@ -0,0 +1,255 @@
...
+```sh
+scrypto test
+```
+
+## Closing thoughts
+
+This section shows how to test a Scrypto package with both unit and integration
+tests, but there's room for more thorough testing even here. We could add more
+tests to cover more edge cases and check for where we should see more failed
+transaction rather than just successful ones. There is also a lot of repeated
```
Comment (motosharpley) [inline step-by-step/19-candy-store-tests/README.md:251]: transactions

**Review by motosharpley — APPROVED** [review]
